adr/2022-06-07-ingress-classes.md

```diff
@@ -0,0 +1,18 @@
+# Ingress classes
+
+Jehoszafat:
+
+To allow network segmentation of the ingress a user can define new ingress classes.
+
+1. An ingress class can be private or public, meaning either a public or private load balancer is used.
+1. Each ingress class is associated with a distinct ingress controller
+1. Each ingress controller is associated with either public or private IP address
+1. There is always the `platform` ingress class defined `otomi.ingress.platform`
+1. It is possible to define additional ingress classes at `otomi.ingress.classes`
+1. All otomi services are by default exposed behind the `platform` ingress class
+
+To allow the keycloak to serve as an authentication gateway for services exposed behind both public and private LBs, the keycloak needs to be assigned to a public ingress class. If `platform` ingress class is private, then an additional public ingress class needs to be created. A public class can be assigned to keycloak at `apps.keycloak.ingressClassName`.
```

Comment:
I am currently not in favor of your "solution" to expose Keycloak, and I want to keep ALL platform services under the same loadbalancer. If a user wants otomi to be private, we allow them to do so. How they access it is then up to their setup. I presume a jumphost is then available to act as a bastion or proxy and the user will set it up accordingly. Out of our scope.

Reply:
It is a bit opposite to what heard from you couple weeks ago, when you insisted that keycloak needs to be accessible to  team services that are public, even if the platform services are chosen to be private. What has change your mind since then?
P.S.
why do you quote word solution ?